SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -0,0 +1,281 @@
+    } else if constexpr (Val == init_val::zero) {
+      return "zero";
+    } else if constexpr (Val == init_val::positive) {
+      return "positive";
+    } else if constexpr (Val == init_val::negative) {
+      return "negative";
+    } else if constexpr (Val == init_val::min_half) {
+      return "min_half";
+    } else if constexpr (Val == init_val::max_half) {
+      return "max_half";
+    } else if constexpr (Val == init_val::neg_inf) {
+      return "neg_inf";
+    } else if constexpr (Val == init_val::nan) {
+      return "nan";
+    } else if constexpr (Val == init_val::denorm) {
+      return "denorm";
+    } else if constexpr (Val == init_val::inexact) {
+      return "inexact";
+    } else if constexpr (Val == init_val::ulp) {
+      return "ulp";
+    } else if constexpr (Val == init_val::ulp_half) {
+      return "ulp_half";
+    } else {
+      static_assert(Val != Val, "Unexpected base value value");
+    }
+  }
+};
+
+template <typename DataT, int NumElems, typename TestCaseT, typename BaseVal,
+          typename Step>
+class run_test {
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    static_assert(std::is_same_v<typename BaseVal::value_type, init_val>,
+                  "BaseVal template parameter should be init_val type.");
+    static_assert(std::is_same_v<typename Step::value_type, init_val>,
+                  "Step template parameter should be init_val type.");
+
+    shared_vector<DataT> result(NumElems, shared_allocator<DataT>(queue));
+
+    const auto base_value = get_value<DataT, BaseVal::value>();
+    const auto step_value = get_value<DataT, Step::value>(base_value);
+
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *const out = result.data();
+
+      cgh.single_task<kernel_for_fill<DataT, NumElems, TestCaseT,
+                                      BaseVal::value, Step::value>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            TestCaseT::template call_simd_ctor<DataT, NumElems>(
+                base_value, step_value, out);
+          });
+    });
+    queue.wait_and_throw();
+
+    bool passed = true;
+
+    DataT expected_value = base_value;
+    for (size_t i = 0; i < result.size(); ++i) {
+      // std::isnan() couldn't be called for integral types because it call is
+      // ambiguous GitHub issue for that case:
+      // https://github.com/microsoft/STL/issues/519
+      if (!are_bitwise_equal(result[i], expected_value)) {
+        if constexpr (type_traits::is_sycl_floating_point_v<DataT>) {
+          if (std::isnan(result[i]) &&
+              (std::isnan(step_value) || std::isnan(base_value))) {
+            continue;
+          }
+        }
+        passed = false;
+
+        const auto description =
+            FillCtorTestDescription<DataT, NumElems, TestCaseT, BaseVal::value,
+                                    Step::value>(i, result[i - 1] + step_value,
+                                                 expected_value, data_type);
```

Comment:
```suggestion
            FillCtorTestDescription<DataT, NumElems, TestCaseT, BaseVal::value,
                                    Step::value>(i, retrieved_value,
                                                 expected_value, data_type);
```
with
```
const auto& retrieved = result[i];
```

Reply:
Now `result[i]` provided into `FillCtorTestDescription` I've done it in this commit: 209a007ff239c59795d58172a05bf77ba4fd5b80